src/plonk/circuit.rs

```diff
@@ -509,12 +510,38 @@ pub trait Assignment<F: Field> {
     fn pop_namespace(&mut self, gadget_name: Option<String>);
 }
 
+/// A floor planning strategy for a circuit.
+///
+/// The floor planner is chip-agnostic and applies its strategy to the circuit it is used
+/// within.
+pub trait FloorPlanner {
+    /// Given the provided `cs`, synthesize the given circuit.
+    ///
+    /// Internally, a floor planner will perform the following operations:
+    /// - Instantiate a [`Layouter`] for this floor planner.
+    /// - Perform any necessary setup or measurement tasks, which may involve one or more
+    ///   calls to `Circuit::default().synthesize(config, &mut layouter)`.
```

Comment:
This should have been `Circuit::without_witnesses().synthesize(...)`.

Reply:
Ah yes, I forgot to update this comment. I'll open a separate PR.